### PR TITLE
Luis Authoring Key

### PR DESCRIPTION
**Pull request by SDESkowronski**
This appears to work for me locally. However, I suspect there might be some issues when run on CircleCI. Please review and provide feedback. 

NOTE: This is my FIRST Node.js app and first time mucking around with CircleCI, so don't assume I know what I am doing here.

### CODE REVIEW

File: .circleci/SetLuisAuthoringKey.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+const envFilePath = '../cl-samples/.env'
+
+const authoringKeys =
+[
+  '11111111111111111111111111111111',
+  '22222222222222222222222222222222',
+  '33333333333333333333333333333333',
+  '44444444444444444444444444444444',
+  '55555555555555555555555555555555',
+  '7d7048bb12ec4bc59c988cfa855363de',
```
[the remark this answers is not in the document]
Comment (SDESkowronski) [reply .circleci/SetLuisAuthoringKey.js]: That last one is my Authoring key. So if we need a new one I'll need it ASAP. But why? Is this PR public?

I don't know how the .env file gets onto the CI machine. It is not under source control, it is listed in .gitignore.

[the remark this answers is not in the document]
Comment (LarsLiden) [reply .circleci/SetLuisAuthoringKey.js]: Yes, this is an open source project so everything is public.  Some nefarious folks write code that automatically scans all open source projects looking for keys that were checked in my mistake, so please rotate right away.  Check with Shahin for how the .env file gets to the CI machine as he set that up

[the remark this answers is not in the document]
Comment (SDESkowronski) [reply .circleci/SetLuisAuthoringKey.js]: I recycled it.

**Comment by mattmazzola** [comment]
Hi guys, came across this email and thought I might try to help since I'm not able to do much else.  The way variables are loaded might have been changed since I've been out, but it used to be set as part of the specific Environment Variables configured on the Circle CI service for this repo. These get set when the process starts and thus are inherited by other process run inside it. There is no .env file in CircleCI. The .env file just facilitates storing environment variables for developers in a standard way so the can be loaded in a standard way and applied to the running process. One way is the `dotenv` package we use.  In this case the dotenv does nothing because there is no file.

1. To modify the environment variables you have to log into CircleCI. If I remember it's associated with our Blis Github team so any contributor can login and edit stuff but maybe it was manual.
2. Go to one of the ConversationLearner-UI builds such as the one above that's failing.
3. Go to "Project Settings" by clicking the gear icon on the far right of the top header. (Image below)

image

4. Go to "Environment Variables" in the "Build Settings" section on the left.

image

There you can see all the variables and enter your "LUIS_AUTHORING_KEY_ALT_1" etc as needed. Don't alter any of the other ones though since they are needed for other things.

Hope that helps.

**Comment by LarsLiden** [comment]
Thanks, Matt!

File: .circleci/SetLuisAuthoringKey.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  process.env.LUIS_AUTHORING_KEY_ALT_3,
+  process.env.LUIS_AUTHORING_KEY_ALT_4,
+  process.env.LUIS_AUTHORING_KEY_ALT_5,
+]
+
+// Randomly pick an authoring key from the array.
+var randomIndex = Math.floor(Math.random() * 5)
+var luisAuthoringKey = authoringKeys[randomIndex]
+
+console.log(`export LUIS_AUTHORING_KEY=${luisAuthoringKey}\n`)
```
[the remark this answers is not in the document]
Comment (SDESkowronski) [reply .circleci/SetLuisAuthoringKey.js]: The log is essential. See `.circleci\config.yml` look for this line...

node .circleci/SetLuisAuthoringKey.js >> $BASH_ENV

**Pull request merged**